metaflow/plugins/kfp/kfp.py

```diff
@@ -393,28 +400,71 @@ def step_op_func(
     print("----")
 
     # TODO: Map username to KFP specific user/profile/namespace
+    # Note: we don't put this in a try catch block as below because
+    # Popen will not produce a error that will cause the Kubeflow step
+    # to stop
     with Popen(
         ["/bin/sh", "-c", cmd],
         stdin=PIPE,
         stdout=PIPE,
-        stderr=STDOUT,
+        stderr=PIPE,
         universal_newlines=True,
         env=dict(os.environ, USERNAME="kfp-user", METAFLOW_RUN_ID=kfp_run_id),
-    ) as process, StringIO() as string_buffer:
+    ) as process, StringIO() as stdout_buffer, StringIO() as stderr_buffer:
         for line in process.stdout:
             print(line, end="")
-            string_buffer.write(line)
-        buffer_output = string_buffer.getvalue()
-    print("___ DONE ___")
+            stdout_buffer.write(line)
+        stdout_output = stdout_buffer.getvalue()
+        for line in process.stderr:
+            print(line, end="")
+            stderr_buffer.write(line)
+        stderr_output = stderr_buffer.getvalue()
+
+    # We put a try-catch block here because if the process above fails,
+    # this line will cause an error which will stop the Kubeflow step.
+    # We want to catch this error so the error logs can be captured
+    # and persisted to s3.
+    # TODO: this behavior is implicit. Any suggestion to make it explicit?
+    try:
+        with open(
+            os.path.join(tempfile.gettempdir(), "kfp_metaflow_out_dict.json"), "r"
+        ) as file:
+            task_out_dict = json.load(file)
+        print("___DONE___")
+    except:
+        print("Error. Persisting error logs to S3.")
```

Comment:
The above exception print statement is incorrect, it's not persisting to S3.
Also, we should catch the explicit exception, is there a file not found exception?

Reply:
Actually this can be simplified by moving this block to be after copying the logs to S3, and after the block:

```python
    if process.returncode != 0:
        raise Exception("Returned: %s" % process.returncode)	        raise Exception("Returned: %s" % process.returncode)
```